elasticdl/python/worker/worker.py

```diff
@@ -268,15 +256,6 @@ def report_prediction_outputs(self, predictions):
             )
         return True
 
-    def _create_variable_and_report(self, features):
-        # Use model.call to create variables, then report to ps
-        _ = self._model.call(features)
-        self._non_embed_vars = get_non_embedding_trainable_vars(
-            self._model, self._embedding_layers
-        )
-        self.report_variable()
-        self._var_created = True
-
```

Comment:
We can't delete `_create_variable_and_report `. Worker should call it (or explicitly build model to create variables) even if it does not need to report variables to master. Without creating trainable variables, model does not have any variable when function `_process_minibatch ` calls function `get_model`, i.e. `model.trainable_variables` is an empty list. `get_model` will do nothing if `model.trainable_variables`.